ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,408 @@
+    end
+    function eval_grad_f_cb(M, X)
+        x = JuMP.vectorize(X, _shape(model.manifold))
+        grad_f = zeros(length(x))
+        if model.sense == MOI.FEASIBILITY_SENSE
+            grad_f .= zero(eltype(grad))
+        else
+            MOI.eval_objective_gradient(evaluator, grad_f, x)
+        end
+        if model.sense == MOI.MAX_SENSE
+            LinearAlgebra.rmul!(grad_f, -1)
+        end
+        reshaped_grad_f = JuMP.reshape_vector(grad_f, _shape(model.manifold))
+        return ManifoldDiff.riemannian_gradient(model.manifold, X, reshaped_grad_f)
+    end
+    mgo = Manopt.ManifoldGradientObjective(eval_f_cb, eval_grad_f_cb)
+    dmgo = decorate_objective!(model.manifold, mgo)
+    model.problem = DefaultManoptProblem(model.manifold, dmgo)
+    reshaped_start = JuMP.reshape_vector(start, _shape(model.manifold))
+    descent_state_type = model.options[DESCENT_STATE_TYPE]
+    kws = Dict{Symbol,Any}(
+        Symbol(key) => value for (key, value) in model.options if key != DESCENT_STATE_TYPE
+    )
+    s = descent_state_type(model.manifold, reshaped_start; kws...)
+    model.state = decorate_state!(s)
+    solve!(model.problem, model.state)
+    return nothing
+end
+
+struct ArrayShape{N} <: JuMP.AbstractShape
+    size::NTuple{N,Int}
+end
+
+function JuMP.vectorize(array::Array{T,N}, ::ArrayShape{M}) where {T,N,M}
+    return vec(array)
+end
+
+function JuMP.reshape_vector(vector::Vector, shape::ArrayShape)
+    return reshape(vector, shape.size)
+end
+
+function _shape(m::ManifoldsBase.AbstractManifold)
+    return ArrayShape(ManifoldsBase.representation_size(m))
+end
+
+function JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
+    shape = _shape(m)
+    return JuMP.VariablesConstrainedOnCreation(
+        JuMP.vectorize(func, shape), VectorizedManifold(m), shape
+    )
+end
+
+function MOI.get(model::Optimizer, ::MOI.TerminationStatus)
+    if isnothing(model.state)
+        return MOI.OPTIMIZE_NOT_CALLED
+    else
+        return MOI.LOCALLY_SOLVED
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.ResultCount)
+    if isnothing(model.state)
+        return 0
+    else
+        return 1
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.PrimalStatus)
+    if isnothing(model.state)
+        return MOI.NO_SOLUTION
+    else
+        return MOI.FEASIBLE_POINT
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.DualStatus) = MOI.NO_SOLUTION
+
+function MOI.get(model::Optimizer, ::MOI.RawStatusString)
+    # `strip` removes the `\n` at the end and returns an `AbstractString`
+    # Since MOI wants a `String`, we pass it through `string`
+    return string(strip(get_reason(model.state)))
+end
+
+function MOI.get(model::Optimizer, attr::MOI.ObjectiveValue)
+    MOI.check_result_index_bounds(model, attr)
+    solution = Manopt.get_solver_return(model.state)
+    return get_cost(model.problem, solution)
+end
+
+function MOI.get(model::Optimizer, attr::MOI.VariablePrimal, vi::MOI.VariableIndex)
```

Comment:
I do understand this signature and idea, but still again a small doc string might be nice?